app/models/recipe.rb

```diff
@@ -1,3 +1,11 @@
 class Recipe < ApplicationRecord
    validates_presence_of :name, :complexity, :genre
+   has_many :recipe_ingredients
+   has_many :ingredients, through: :recipe_ingredients
+
+   def total_cost
+      ingredients.sum do |ingredient|
```

Comment:
We'd like to see you do these calculations with AR instead of Ruby, and most of the time it's easier that way! Consider this instead:
`ingredients.sum(:cost)`